internal/command/metadata_functions_test.go

```diff
@@ -0,0 +1,35 @@
+package command
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/mitchellh/cli"
+)
+
+func TestMetadataFunctions_error(t *testing.T) {
+	ui := new(cli.MockUi)
+	c := &MetadataFunctionsCommand{
+		Meta: Meta{
+			Ui: ui,
+		},
+	}
+
+	if code := c.Run(nil); code != 1 {
+		fmt.Println(ui.ErrorWriter.String())
+		t.Fatalf("expected error: \n%s", ui.ErrorWriter.String())
+	}
+}
+
+func TestMetadataFunctions_output(t *testing.T) {
+	ui := new(cli.MockUi)
+	m := Meta{Ui: ui}
+	c := &MetadataFunctionsCommand{Meta: m}
+
+	if code := c.Run([]string{"-json"}); code != 0 {
+		t.Fatalf("wrong exit status %d; want 0\nstderr: %s", code, ui.ErrorWriter.String())
+	}
+
+	// gotString := ui.OutputWriter.String()
+	// TODO how to mock scope.Functions() to reduce the output length
```

Comment:
This might be possible with `MockEvalContext`, but unnecessary: I think this test should assert that 
 - command exits successfully
 - command does not print anything to stderr
 - command prints a string to stdout that is valid JSON
 - the JSON has the expected structure and contains at least N functions (for a reasonable N)
 - (perhaps) the data for a particular stable function (e.g. `max`) is correct

If all these things are true, I think the main risk is that the list of functions contains one that cannot be correctly marshalled, but no error is thrown, and it is just missed out of the JSON. This scenario can be dealt with in `TestMarshalFunction`.

Reply:
Thanks for the suggestions, I've updated the test accordingly.